docs/cody/clients/model-configuration.mdx

```diff
@@ -0,0 +1,520 @@
+      // Only allow "beta" and "stable" models.
+      // Not "experimental" or "deprecated".
+      "statusFilter": ["beta", "stable"],
+
+      // Allow any models provided by Anthropic or OpenAI.
+      "allow": [
+        "anthropic::*",
+        "openai::*"
+      ],
+
+      // Do not include any models with the Model ID containing "turbo",
+      // or any from AcmeCo.
+      "deny": [
+        "*turbo*",
+        "acmeco::*"
+      ]
+    }
+  }
+}
+```
+
+## Default Models
+
+The `"modelConfiguration"` setting also contains a `"defaultModels"` field that allows you to specify the LLM model used depending on the situation. If no default is specified, or refers to a model that isn't found, it will silently fallback to a suitable alternative.
+
+You can leave these settings blank and have reasonable defaults be chosen for you, or you can express the models explicitly.
+
+```json
+  ...
+  "modelConfiguration": {
+    "defaultModels": {
+      "chat": "anthropic::2023-06-01::claude-3.5-sonnet",
+      "codeCompletion": "anthropic::2023-06-01::claude-3.5-sonnet",
+      "fastChat": "anthropic::2023-06-01::claude-3-haiku"
+    }
+  }
+  ...
+```
+
+The format of these strings is a "Model Reference", which is a format for uniquely identifying each LLM model exposed from your Sourcegraph instance.
+
+## Advanced Configruation
+
+For most administrators, relying on the LLM models made available by Cody Gateway is sufficient. However, if even more customization is required, you can configure your own LLM providers and models.
+
+<Callout type="warning">Defining your own LLM providers and models is an advanced use-case and requires care to get the correct results. It also may bypass protections to ensure compatibiloity between your Sourcegraph instance and LLMs. If you need help contact your Sourcegraph account executive.</Callout>
+
+### Overview
+
+The `"modelConfiguration"` section exposes two fields `"providerOverrides"` and `"modelOverrides"`. These may override any Sourcegraph-supplied data, or simply introduce new ones entirely.
+
+### Provider Configuration
+
+A "provider" a way to organize LLM models. Typically a provider would be referring to the company that produced the model. Or the specific API/service being used to access the model. But conceptually, it's just a namespace.
+
+By defining a provider override in your Sourcegraph site configuration, you are introducing a new namespace to contain models. Or customize the existing provider namespace supplied by Sourcegraph. (e.g. all `"anthropic"` models.)
+
+The following configuration shippet defines a single provider override with the ID `"anthropic-direct"`.
+
+```json
+"modelConfiguration": {
+  // Do not use any Sourcegraph-supplied models.
+  "sourcegraph": null,
+
+  // Introduce our own, custom provider "anthropic-direct".
+  "providerOverrides": [
+    {
+      "id": "anthropic-direct",
+      "displayName": "Anthropic models, sent directly to anthropic.com",
```

Comment:
This display name shows up on the logs and only the logs, right?

Reply:
Currently, I don't know if we show the display name _anywhere_. The provider _ID_ will show up in logs (since it is baked into the model references.)

The intent of adding a separate display name field was so that we could display this human-friendly text to users, e.g. when differentiating between various flavors of Claude in the Cody IDE extension.

Reply:
Got it - thanks for confirming Chris!